### ISSUE TITLE
Mongo Query Selectors are flagged as invalid.

### ISSUE THREAD

**Issue by brucejo75** [meteor/meteor#8775]
pr #8608 disallows all [query selectors](https://docs.mongodb.com/manual/reference/operator/query/).

Here is the problem: [modify.js#L251](https://github.com/meteor/meteor/blob/devel/packages/minimongo/modify.js#L251)

I ran a simple test:
```
// set the LocalData collection
LocalData =  new Mongo.Collection(null);

function setGlobalContext() {
  let gContext = {'globalContext': {}};
  gContext.globalContext.test = 'test';
  LocalData.upsert({'globalContext': {'$exists': true}}, gContext);
}

setGlobalContext();
```

the code throws because `$exists` has a `$` character.

I think tests need to be updated?

**Comment by benjamn** [comment]
@hwillson Thoughts?

**Comment by hwillson** [comment]
Ouch! Findings:

- Query selectors work properly with `update` (non-upsert), `find` / `findOne`
- Query selectors work with `upsert` when the record being upserted already exists
- Query selectors fail when being used with `upsert` (or an upsert `update`) and the record being upserted doesn't exist

So yes, definitely a bug and yes the tests are missing this 🙁.

I'll prepare a patch right now. Thanks for pointing this out @brucejo75!

**Comment by sebakerckhof** [comment]
Okay, this should be fixed in next 1.6 build now that https://github.com/meteor/meteor/issues/8815 is merged.

**Comment by hwillson** [comment]
Great, closing for now - thanks for your work on this @sebakerckhof!
